src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -656,6 +657,19 @@ int sd_dhcp_client_set_fallback_lease_lifetime(sd_dhcp_client *client, uint64_t
         return 0;
 }
 
+int sd_dhcp_client_set_bootp(sd_dhcp_client *client, int bootp) {
+        assert_return(client, -EINVAL);
+        assert_return(!sd_dhcp_client_is_running(client), -EBUSY);
+
+        client->bootp = bootp;
+
+        /* For BOOTP mode, we don't want to send any request options by default. */
+        set_free(client->req_opts);
+        client->req_opts = NULL;
```

Comment:
I don't completely agree with this assertion. I need to spend more time understanding what options might actually be here or not. I agree that we should not send any DHCP options, but there may be legitimate BOOTP options that we should not strip out.